File: python_master_reference/18_functions.py
"""
18 - Functions
A function is a reusable block of code.
"""

def greet():
    print("Hello from a function!")

greet()


def greet_user(name):
    print(f"Hello, {name}!")

greet_user("Khadir")


def add_numbers(num1, num2):
    result = num1 + num2
    return result

sum_result = add_numbers(5, 7)

print(sum_result)


def login(username, password):
    print(f"Typing username: {username}")
    print(f"Typing password: {password}")
    print("Clicking login button")

login("admin@juice-sh.op", "admin123")

File: typescript_master_reference/18_functions.ts
/*
18 - Functions
A function is a reusable block of code.
*/
function greet(): void {
  console.log("Hello from a function!");
}
greet();

function greetUser(name: string): void {
  console.log(`Hello, ${name}!`);
}
greetUser("Khadir");

function addNumbers(num1: number, num2: number): number {
  const result = num1 + num2;
  return result;
}
const sumResult = addNumbers(5, 7);
console.log(sumResult);

function login(username: string, password: string): void {
  console.log(`Typing username: ${username}`);
  console.log(`Typing password: ${password}`);
  console.log("Clicking login button");
}
login("admin@juice-sh.op", "admin123");
